### DISCUSSION TITLE
What features will the new map format and editor support?

### DISCUSSION THREAD

**Discussion by Mr-Bajs** [ihhub/fheroes2#8087] [General]
Apart from being a map editor with the features of the original editor, what new features to the dev team have in mind?

A few features that comes to mind would be bit more metadata for the maps.
Data such as far as i know isnt included in the original mapformat:

- **Map version** - Versioning would be great it se if a map has been improved upon, with new features, rebalanced etc. 
- **Date for last change** - Could be intresting when the last change to the map was made.
- **Licence** - under what license the maps is created. Would be nice to have the ability to change and improve maps and redestribute them, especially if they have any know bugs or design flaws, if they are under a free license then it would be no problem at least no problem to know if you can. An free license should be the default.
- **Author** - Ofcource the authors should be included in the metadata.

Data should be accessible in the editor, though I could see a point of having map version when you select a scenario.

**Translation**
It would be awesome if fheroes2 maps format would have the maps translatable, preferable in the file format itself, so a user just have to handle one file for each map.

Also if translations support would be added then, supported languages would be nice if it where included in the map metadata.

**Comment by ihhub** [comment]
We will take these into an account. There will be no license for maps. All products made by this software are automatically under the same license.

**Reply by oleg-derevenetz** [reply]
Hi @ihhub

> All products made by this software are automatically under the same license.

I'm not sure that there is a way to enforce the license on maps created by our editor (unless we make some special EULA for it, but this approach is also doubtful IMHO, despite the fact that the original HoMM2 uses a similar approach):

https://www.gnu.org/licenses/gpl-faq.html#WhatCaseIsOutputGPL

**Reply by ihhub** [reply]
Hi @oleg-derevenetz , you are right. My main point is that since our product then all maps should be free as well.

**Reply by oleg-derevenetz** [reply]
@ihhub In addition, let's assume that the creator of the map wants to use a non-standard license. In this case, should we include the full license text in the map file? I think it's hardly necessary.

**Reply by ihhub** [reply]
As a map creator in the past I can say that a map can be distributed together with a license file if needed. We shouldn't handle license issues for maps. At a very least we can include some kind of an EULA file stating default license for any maps, unless the author decides to change it.

**Reply by Mr-Bajs** [reply]
I think to have the maps should be released under the same license as fheroes2 by default, GPL-2.0 license and the license should come with the map for clarity. If the license are there from the start then most of the risk for any future headache are mitigated.

However i don't think it's any way necessary to support any other licenses. If someone really wanted to release a map under another license, they are free to go through the hassle change the mapfile themselves. How they do it are up to them if they would want to.
